### PR TITLE
Highlight the DisplayText and not the FilterText when highlighting completion items.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@vasily-kirichenko

Can you take a look and see if this addresses: https://github.com/dotnet/roslyn/issues/17969

**Comment by vasily-kirichenko** [comment]
@CyrusNajmabadi I don't know. Have you covered the following cases:

* DisplayText = Choice.bind and FilterText = "bind" => Choice.**bind**
* DisplayText = Choice.bind (open N1.N2) and FilterText = bind => Choice.**bind** (open N1.N2)
* DisplayText = Choice.bind (open N1.bind.N2.bind2.N3) and FilterText = bind => Choice.**bind** (open N1.bind.N2.bind2.N3)
* DisplayText = Choice.bind1.bind2.bind3 and FilterText = bind => Choice.bind1.bind2.**bind**3
* DisplayText = Choice.bind.bind1 (open bind) and FilterText = bind => Choice.bind.**bind**1 (open bind)

?

**Comment by CyrusNajmabadi** [comment]
I have no idea :)   You're the one making completion lists with those items.  So i was hoping you could try this out against that to see :)

**Comment by vasily-kirichenko** [comment]
Is it possible to grab compiled assembly and put it somewhere so that Exp VS instance load it?

**Comment by CyrusNajmabadi** [comment]
I just open Roslyn.sln, and f5 the VisualStudioSetup.next project:

image

This will launch an experimental hive with the built roslyn bits.

**Comment by CyrusNajmabadi** [comment]
Goign to bed now.  Let me know how this works out for you!

**Comment by vasily-kirichenko** [comment]
I see. But I need to load Visual F# Tools built from my branch into that exp instance, too :) OK, thanks, will try.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski @Pilchie Is there a way for @vasily-kirichenko to dogfood both Roslyn and F# bits easily?

**Comment by Pilchie** [comment]
Here - `https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/Setup/VisualStudioSetup.csproj#L14` is where we set our bits to go to `RoslynDev` (along with the same in other projects that make VSIXs).  You can change either Roslyn to match F# or Vice-versa and then just build both.

**Comment by CyrusNajmabadi** [comment]
@vasily-kirichenko Did this help?

**Comment by vasily-kirichenko** [comment]
@CyrusNajmabadi I've not had a chance to try it. I think I'm gonna wait until you guys release the fix in a nuget package.

**Comment by CyrusNajmabadi** [comment]
it's not going to be released until we can verify it actually fixes the issue you're seeing :)

**Comment by vasily-kirichenko** [comment]
Dead lock? ;) OK, I'll try to test it, but not sooner than this weekend.

**Comment by vasily-kirichenko** [comment]
cibuld.cmd, restore.cmd do not work behind proxy.
```
λ cibuild.cmd
New Args are
Parameters:
Downloading NuGet.exe
System.Net.WebException: The remote server returned an error: (407) Proxy Authentication Required.
   at System.Net.WebClient.DownloadFile(Uri address, String fileName)
   at CallSite.Target(Closure , CallSite , Object , String , Object )
Downloading NuGet.exe
System.Net.WebException: The remote server returned an error: (407) Proxy Authentication Required.
   at System.Net.WebClient.DownloadFile(Uri address, String fileName)
   at CallSite.Target(Closure , CallSite , Object , String , Object )
The term 'E:\github\roslyn\NuGet.exe' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
System.Management.Automation.CommandNotFoundException: The term 'E:\github\roslyn\NuGet.exe' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
```
```
λ Restore.cmd
New Args are
Downloading NuGet.exe
System.Net.WebException: The remote server returned an error: (407) Proxy Authentication Required.
   at System.Net.WebClient.DownloadFile(Uri address, String fileName)
   at CallSite.Target(Closure , CallSite , Object , String , Object )
Downloading NuGet.exe
System.Net.WebException: The remote server returned an error: (407) Proxy Authentication Required.
   at System.Net.WebClient.DownloadFile(Uri address, String fileName)
   at CallSite.Target(Closure , CallSite , Object , String , Object )
Downloading NuGet.exe
System.Net.WebException: The remote server returned an error: (407) Proxy Authentication Required.
   at System.Net.WebClient.DownloadFile(Uri address, String fileName)
   at CallSite.Target(Closure , CallSite , Object , String , Object )
The term 'E:\github\roslyn\NuGet.exe' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```
As I understand, it's not possible to load the solution into VS without ^^^ been successfully finished:

image

```
E:\github\roslyn\src\Workspaces\Remote\Core\RemoteWorkspaces.csproj : error  : The imported project "C:\Users\kirichenko\.nuget\packages\Microsoft.VSSDK.BuildTools\15.0.26124-RC3\build\Microsoft.VsSDK.BuildTools.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.  E:\github\roslyn\build\Targets\Settings.props

E:\github\roslyn\src\Workspaces\Remote\ServiceHub\ServiceHub.csproj : error  : The imported project "C:\Users\kirichenko\.nuget\packages\Microsoft.VSSDK.BuildTools\15.0.26124-RC3\build\Microsoft.VsSDK.BuildTools.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.  E:\github\roslyn\build\Targets\Settings.props

E:\github\roslyn\src\Tools\RepoUtil\RepoUtil.csproj : error  : The imported project "C:\Users\kirichenko\.nuget\packages\Microsoft.VSSDK.BuildTools\15.0.26124-RC3\build\Microsoft.VsSDK.BuildTools.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.  E:\github\roslyn\build\Targets\Settings.props

(and so on, so on)
```

**Comment by Pilchie** [comment]
@vasily-kirichenko How do you use NuGet in general?

@jaredpar any ideas here?

**Comment by vasily-kirichenko** [comment]
OK, nuget works at home, so I've built roslyn and F5 works. I set exp hive name to FSharpDev, same as in Visual F# Tool, build again, then build VFT and F5. After loading a F# solution, VS crashes. Debugger shows this:

image

**Comment by vasily-kirichenko** [comment]
VFT uses Microsoft.CodeAnalysis.xxx assemblies from NuGet packages. I'm afraid that in order to test this I need somehow point VFT to freshly built Roslyn binaries. But how?

**Comment by vasily-kirichenko** [comment]
I've copied dlls from `Binaries\Debug\Dlls` to corresponding nuget packages in VFT solution and rebuild it. Same error.

**Comment by Pilchie** [comment]
Can you get the callstack and exception details?

Also, did you update all of the vsixmanifest files under `src` in Roslyn.sln?

**Comment by vasily-kirichenko** [comment]
```
>	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.ErrorReporting.FatalError.Report(System.Exception exception = Count = 1, System.Action<System.Exception> handler = {System.Action<System.Exception>}) Line 151	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Roslyn.Utilities.TaskExtensions.ReportFatalErrorWorker(System.Threading.Tasks.Task task = Id = 4022, Status = Faulted, Method = "{null}", Result = "{Not yet computed}", object continuationFunction = {System.Func<System.Threading.Tasks.Task>}) Line 395	C#
 	mscorlib.dll!System.Threading.Tasks.ContinuationTaskFromTask.InnerInvoke()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.Execute()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.ExecutionContextCallback(object obj)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.ExecuteWithThreadLocal(ref System.Threading.Tasks.Task currentTaskSlot = Id = 4023, Status = Running, Method = "Void ReportFatalErrorWorker(System.Threading.Tasks.Task, System.Object)")	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.ExecuteEntry(bool bPreventDoubleExecution)	Unknown
 	mscorlib.dll!System.Threading.Tasks.ThreadPoolTaskScheduler.TryExecuteTaskInline(System.Threading.Tasks.Task task, bool taskWasPreviouslyQueued = false)	Unknown
 	mscorlib.dll!System.Threading.Tasks.TaskScheduler.TryRunInline(System.Threading.Tasks.Task task = Id = 4023, Status = Running, Method = "Void ReportFatalErrorWorker(System.Threading.Tasks.Task, System.Object)", bool taskWasPreviouslyQueued)	Unknown
 	mscorlib.dll!System.Threading.Tasks.TaskContinuation.InlineIfPossibleOrElseQueue(System.Threading.Tasks.Task task = Id = 4023, Status = Running, Method = "Void ReportFatalErrorWorker(System.Threading.Tasks.Task, System.Object)", bool needsProtection)	Unknown
 	mscorlib.dll!System.Threading.Tasks.StandardTaskContinuation.Run(System.Threading.Tasks.Task completedTask, bool bCanInlineContinuationTask)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishStageThree()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishStageTwo()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.Finish(bool bUserDelegateExecuted)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task<System.Threading.Tasks.TaskExtensions.VoidResult>.TrySetException(object exceptionObject)	Unknown
 	mscorlib.dll!System.Threading.Tasks.UnwrapPromise<System.Threading.Tasks.TaskExtensions.VoidResult>.TrySetFromTask(System.Threading.Tasks.Task task, bool lookForOce)	Unknown
 	mscorlib.dll!System.Threading.Tasks.UnwrapPromise<System.Threading.Tasks.TaskExtensions.VoidResult>.InvokeCore(System.Threading.Tasks.Task completingTask)	Unknown
 	mscorlib.dll!System.Threading.Tasks.UnwrapPromise<System.Threading.Tasks.TaskExtensions.VoidResult>.Invoke(System.Threading.Tasks.Task completingTask)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishStageThree()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishStageTwo()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.Finish(bool bUserDelegateExecuted)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task<System.Threading.Tasks.VoidTaskResult>.TrySetException(object exceptionObject)	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder<System.Threading.Tasks.VoidTaskResult>.SetException(System.Exception exception)	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder.SetException(System.Exception exception)	Unknown
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.SolutionCrawler.IdleProcessor.ProcessAsync() Line 76	C#
 	[Resuming Async Method]	
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.InvokeMoveNext(object stateMachine)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.Run()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.OutputAsyncCausalityEvents.AnonymousMethod__0()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.ContinuationWrapper.Invoke()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.TaskAwaiter.OutputWaitEtwEvents.AnonymousMethod__0()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.ContinuationWrapper.Invoke()	Unknown
 	mscorlib.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(System.Action action, bool allowInlining, ref System.Threading.Tasks.Task currentTask = Id = 4023, Status = Running, Method = "Void ReportFatalErrorWorker(System.Threading.Tasks.Task, System.Object)")	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishStageThree()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task<System.Threading.Tasks.VoidTaskResult>.TrySetResult(System.Threading.Tasks.VoidTaskResult result)	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder<System.Threading.Tasks.VoidTaskResult>.SetResult(System.Threading.Tasks.VoidTaskResult result)	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder.SetResult()	Unknown
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.SolutionCrawler.IdleProcessor.WaitForIdleAsync() Line 72	C#
 	[Resuming Async Method]	
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.InvokeMoveNext(object stateMachine)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.Run()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.OutputAsyncCausalityEvents.AnonymousMethod__0()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.ContinuationWrapper.Invoke()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.TaskAwaiter.OutputWaitEtwEvents.AnonymousMethod__0()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.ContinuationWrapper.Invoke()	Unknown
 	mscorlib.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(System.Action action, bool allowInlining, ref System.Threading.Tasks.Task currentTask = Id = 4023, Status = Running, Method = "Void ReportFatalErrorWorker(System.Threading.Tasks.Task, System.Object)")	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishStageThree()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task<System.Threading.Tasks.VoidTaskResult>.TrySetResult(System.Threading.Tasks.VoidTaskResult result)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.DelayPromise.Complete()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.Delay.AnonymousMethod__274_1(object state)	Unknown
 	mscorlib.dll!System.Threading.TimerQueueTimer.CallCallbackInContext(object state)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.TimerQueueTimer.CallCallback()	Unknown
 	mscorlib.dll!System.Threading.TimerQueueTimer.Fire()	Unknown
 	mscorlib.dll!System.Threading.TimerQueue.FireNextTimers()	Unknown
 	mscorlib.dll!System.Threading.TimerQueue.AppDomainTimerCallback()	Unknown
```

**Comment by Pilchie** [comment]
Err, not vsixmanifest, but hive locations I mean.

**Comment by Pilchie** [comment]
and $exception.StackTrace?

**Comment by vasily-kirichenko** [comment]
image

**Comment by vasily-kirichenko** [comment]
```
   at Microsoft.VisualStudio.LanguageServices.Remote.JsonRpcSession.<InitializeAsync>d__7.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)
   at Microsoft.VisualStudio.LanguageServices.Remote.JsonRpcSession.InitializeAsync()
   at Microsoft.VisualStudio.LanguageServices.Remote.JsonRpcSession.<CreateAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.LanguageServices.Remote.ServiceHubRemoteHostClient.<CreateServiceSessionAsync>d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.CodeAnalysis.Remote.RemoteHostClient.<TryCreateServiceSessionAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.LanguageServices.Remote.ServiceHubRemoteHostClient.WorkspaceHost.<RegisterPrimarySolutionAsync>d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.LanguageServices.Remote.ServiceHubRemoteHostClient.<RegisterWorkspaceHostAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.LanguageServices.Remote.ServiceHubRemoteHostClient.<CreateAsync>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.LanguageServices.Remote.RemoteHostClientFactory.<CreateAsync>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.VisualStudio.LanguageServices.Remote.RemoteHostClientServiceFactory.RemoteHostClientService.<EnableAsync>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.VisualStudio.LanguageServices.Remote.RemoteHostClientServiceFactory.SolutionChecksumUpdater.<SynchronizePrimaryWorkspaceAsync>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.VisualStudio.LanguageServices.Remote.RemoteHostClientServiceFactory.SolutionChecksumUpdater.<ExecuteAsync>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.CodeAnalysis.SolutionCrawler.IdleProcessor.<ProcessAsync>d__12.MoveNext()
```

**Comment by vasily-kirichenko** [comment]
image

**Comment by vasily-kirichenko** [comment]
https://drive.google.com/open?id=0B8HLQUKik9VtU01WNUZuak1YWjA

**Comment by vasily-kirichenko** [comment]
https://drive.google.com/open?id=0B8HLQUKik9Vtb0F5ZmwzY0RDVlU

**Comment by CyrusNajmabadi** [comment]
@vasily-kirichenko Did you have any luck trying this out ?  If not, i can attempt to synthesize the types of scenarios you have in your examples.

**Comment by CyrusNajmabadi** [comment]
UI looks like this now:

image

**Comment by vasily-kirichenko** [comment]
@CyrusNajmabadi Thanks, it looks correctly on your screenshot.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide I need review on this.

File: src/Features/Core/Portable/Completion/CompletionHelper.cs [hunk 1]
```diff
@@ -49,58 +48,61 @@ public static CompletionHelper GetHelper(Document document)
...
             return match == null ? ImmutableArray<TextSpan>.Empty : match.Value.MatchedSpans;
         }
 
         /// <summary>
-        /// Returns true if the completion item matches the filter text typed so far.  Returns 'true'
+        /// Returns true if the completion item matches the pattern so far.  Returns 'true'
         /// iff the completion item matches and should be included in the filtered completion
         /// results, or false if it should not be.
         /// </summary>
-        public bool MatchesFilterText(CompletionItem item, string filterText, CultureInfo culture)
```
Comment (Pilchie) [inline src/Features/Core/Portable/Completion/CompletionHelper.cs]: Does F#/TS/etc call this API directly?

File: src/Features/Core/Portable/Completion/CompletionHelper.cs [hunk 2]
```diff
@@ -49,58 +48,61 @@ public static CompletionHelper GetHelper(Document document)
...
-        private PatternMatch? GetMatch(CompletionItem item, string filterText, CultureInfo culture)
-        {
-            return GetMatch(item, filterText, includeMatchSpans: false, culture: culture);
-        }
+        private PatternMatch? GetMatch(string text, string pattern, CultureInfo culture)
+            => GetMatch(text, pattern, includeMatchSpans: false, culture: culture);
 
         private PatternMatch? GetMatch(
-            CompletionItem item, string filterText,
+            string completionItemText, string pattern,
```
Comment (Pilchie) [inline src/Features/Core/Portable/Completion/CompletionHelper.cs]: How come the private helper names it `completionItemText`, but everything else uses just `text`.

File: src/Features/Core/Portable/Completion/CompletionHelper.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Completion/CompletionHelper.cs]: I've checked, and neither languages call these directly.

**Pull request merged**